internal/rule/condition.go

```diff
@@ -3,82 +3,111 @@ package rule
 import (
 	"fmt"
 	"github.com/icinga/icinga-notifications/internal/event"
+	"github.com/icinga/icinga-notifications/internal/filter"
+	"math"
 	"time"
 )
 
+// RetryNever indicates that an escalation condition should never be retried once it has been evaluated.
+const RetryNever = time.Duration(math.MaxInt64)
+
 type EscalationFilter struct {
 	IncidentAge      time.Duration
 	IncidentSeverity event.Severity
 }
 
-func (c *EscalationFilter) EvalEqual(key string, value string) (bool, error) {
+// ExtractRetryAfter extracts a time.Duration from the given filter conditions.
+// The retry after duration is extracted from the specified filter conditions that evaluate the "incident_age"
+// filter column. This function returns rule.RetryNever when all the incident age filter values are less than
+// the actual incident age or none of the filter conditions evaluates the "incident_age" column. Otherwise,
+// a time.Duration < rule.RetryNever and > the current incident age at which this escalation condition should
+// be re-evaluated is returned.
+func (e *EscalationFilter) ExtractRetryAfter(conditions []filter.Conditioner, incidentAge time.Duration) time.Duration {
```

Comment:
This should either not be implemented on `*EscalationFilter` or not have the `incidentAge` parameter.